sklearn/metrics/_classification.py

```diff
@@ -2953,17 +2956,16 @@ def log_loss(y_true, y_pred, *, normalize=True, sample_weight=None, labels=None)
     ...          [[.1, .9], [.9, .1], [.8, .2], [.35, .65]])
     0.21616...
     """
-    y_pred = check_array(
-        y_pred, ensure_2d=False, dtype=[np.float64, np.float32, np.float16]
-    )
+    xp, _, device_ = get_namespace_and_device(y_true, y_pred, sample_weight, labels)
+    y_pred = check_array(y_pred, ensure_2d=False, dtype=supported_float_dtypes(xp=xp))
 
     check_consistent_length(y_pred, y_true, sample_weight)
     lb = LabelBinarizer()
 
     if labels is not None:
-        lb.fit(labels)
+        lb.fit(_convert_to_numpy(labels, xp=xp))
```

Comment:
This is somewhat then missing the point of "supporting array API" here. I'd say we support array API if we don't convert to Numpy, and here we do. So in effect, there's not much of an improvement with this PR.

I think in order to get this merged, `LabelBinarizer` should support array API.

Reply:
It might not be worth moving the data back and forth between devices

Reply:
I think that means we need to shelf this PR until we fix label binarizer.